runtime/lua/json.lua

```diff
@@ -0,0 +1,383 @@
+--
+-- json.lua
+--
+-- Copyright (c) 2015 rxi
+--
+-- This library is free software; you can redistribute it and/or modify it
+-- under the terms of the MIT license. See LICENSE for details.
+--
+-- Taken from: https://github.com/rxi/json.lua
+--
+
+local json = { _version = "0.1.0" }
```

Comment:
One very important think to take into account: if this is just an implementation detail (and say we'd use the native nvim functionality later), we must do our best to hide this from other Lua plugins.

Otherwise, people will start relying on it and we will forever be forced to carry some external Lua plugin inside of our runtime folder, which may not be desirable.

Also, suppose there are small inconsistencies between the ways in which this module serialized json vs. native nvim, then users will be scratching their heads as to why it isn't the same. Which doesn't seem ideal.